### PR TITLE
[web] Add shared tests for ui.Path

### PR DESCRIPTION
**Pull request by hterkelsen**
Adds shared tests for ui.Path to the UI test suite.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on writing and running engine tests.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: lib/web_ui/lib/src/engine/canvaskit/path_metrics.dart [hunk 1]
```diff
@@ -20,9 +20,11 @@ class CkPathMetrics extends IterableBase<ui.PathMetric>
 
   /// The [CkPath.isEmpty] case is special-cased to avoid booting the WASM machinery just to find out there are no contours.
   @override
-  Iterator<ui.PathMetric> get iterator => _path.isEmpty
+  Iterator<ui.PathMetric> get iterator => _iterator ??= _path.isEmpty
       ? const CkPathMetricIteratorEmpty._()
       : CkContourMeasureIter(this);
+
+  Iterator<ui.PathMetric>? _iterator;
```
Comment (mdebbar) [inline lib/web_ui/lib/src/engine/canvaskit/path_metrics.dart:27]: Can this become a `late final`?
```suggestion
  late final Iterator<ui.PathMetric> iterator = _path.isEmpty
      ? const CkPathMetricIteratorEmpty._()
      : CkContourMeasureIter(this);
```

**Review by mdebbar — APPROVED** [review]
Call `setUpUiTest()` in `lib/web_ui/test/ui/title_test.dart`?

File: lib/web_ui/lib/src/engine/canvaskit/path_metrics.dart (hunk 1 above)
Comment (hterkelsen) [reply lib/web_ui/lib/src/engine/canvaskit/path_metrics.dart:27]: done

**Comment by hterkelsen** [comment]
> Call `setUpUiTest()` in `lib/web_ui/test/ui/title_test.dart`?

done

**Comment by skia-gold** [comment]
Gold has detected about 6 new digest(s) on patchset 5.
View them at https://flutter-engine-gold.skia.org/cl/github/37222

**Comment by skia-gold** [comment]
Gold has detected about 6 new digest(s) on patchset 6.
View them at https://flutter-engine-gold.skia.org/cl/github/37222

**Pull request merged**